File: lang/python/hello.py
#!/usr/bin/env python

import argparse

def main():
    parser = argparse.ArgumentParser(description="Hello program")
    parser.add_argument("-g", "--greeting",
                        action="store",
                        default="Hello, world!",
                        metavar="TEXT",
                        help="Output text instead of the default greeting."),
    parser.add_argument("-t", "--traditional",
                        action="store_true",
                        default=False,
                        help="Output the traditional greeting message 'hello, world'.")
    parser.add_argument('--version', action='version', version='%(prog)s 1.0')

    opt = parser.parse_args()

    text = "hello, world" if opt.traditional else opt.greeting

    print(text)


if __name__ == '__main__': main()

File: lang/ruby/hello.rb
#!/usr/bin/env ruby

require 'optparse'

text = "Hello, World!"

OptionParser.new do |opt|
  opt.banner = "Usage: #{opt.program_name} [options]"
  opt.version = "1.0"

  opt.on("-g", "--greeting=TEXT") {|newtext|
    text = newtext
  }
  opt.on("-t", "--traditional") {
    text = "hello, world"
  }

  opt.parse!(ARGV)
end

puts text
